native windows build fixes

kcking opened the pull request:
Hi there! I am using the mediasoup crate on native windows and ran into some issues with the Makefile / build.rs script.

I made the following changes

1. use windows `rmdir` instead of `rm`
2. use backslashes in file paths since `rmdir` can't handle forward slashes
3. move comments to start-of-line so they are not interpreted as part of shell command
4. remove clean-all step from build.rs (note, `MEDIASOUP_OUT_DIR` was not being set in this `Command`, so it _appears_ to be a no-op currently)

Let me know if these are changes you would consider merging upstream. For now they can just live on my personal fork :)

kcking committed:
make Makefile more native windows-compatible

kcking committed:
use backslashes in makefile invocation

kcking committed:
remove cleaning which was deleting compiled libraries

ibc commented:
I just wonder if current build system is designed to support native Windows, and also wonder whether these changes here may break other working platforms, but I have no idea about Windows. @nazar-pc thoughts?

nazar-pc commented:
Wasn't designed (especially since you'd need MinGW for Makefile anyway), but I'll check these changes in more detail later.
If it doesn't break things, I have nothing against supporting more configurations of course.

kcking commented:
Thanks for the quick reply!

>  you'd need MinGW for Makefile anyway

Just to be clear I'm actually using `scoop`-installed make with these changes, so no need for mingw as far as I can tell.

```
# cmd.exe
C:\Users\kevin\...>where make
C:\Users\kevin\scoop\shims\make.exe
C:\Users\kevin\...>make --version
GNU Make 4.3
Built for Windows32
```

nazar-pc commented:
Well, then it breaks other environments as you can see from the failed CI.

nazar-pc commented:
If you have time working on #701 would be more productive than work around various kinds of `make` available on Windows that all behave differently.

nazar-pc commented on worker/build.rs:
```diff
@@ -93,20 +92,6 @@ fn main() {
         {
             panic!("Failed to build libmediasoup-worker")
         }
-
-        if env::var("KEEP_BUILD_ARTIFACTS") != Ok("1".to_string()) {
-            // Clean
-            if !Command::new("make")
-                .arg("clean-all")
-                .spawn()
-                .expect("Failed to start")
-                .wait()
-                .expect("Wasn't running")
-                .success()
-            {
-                panic!("Failed to clean libmediasoup-worker")
-            }
-        }
```
This is important for package publishing for instance. But yes, `MEDIASOUP_OUT_DIR` is missing here and needs to be added.

kcking committed:
fix more windows paths

kcking commented:
> Well, then it breaks other environments as you can see from the failed CI.

I think this is just because I didn't convert a couple of other paths to backslashes. I pushed a commit that fixes those as well , but totally understand if you're not looking to accomodate native windows make.

> If you have time working on #701 would be more productive than work around various kinds of make available on Windows that all behave differently.

That's fair. Unfortunately I don't expect to have the time to do that any time soon.

Thanks again for looking at the PR! I can push a fix for the clean-all step in a separate PR.

nazar-pc commented:
> I can push a fix for the clean-all step in a separate PR.

Please do

kcking commented:
> > I can push a fix for the clean-all step in a separate PR.
> 
> Please do

Here you go https://github.com/versatica/mediasoup/pull/726

kcking commented:
https://github.com/versatica/mediasoup/runs/4412988116?check_suite_focus=true#step:8:147 🤦 thanks for trying haha

It's now more wrong but more consistent ✨

kcking committed:
try removing -C

git build on windows is failing, and it appears the build script is
modifying subprojects in-tree

nazar-pc commented on worker/Makefile:
```diff
@@ -134,7 +134,7 @@ clean-all: clean-subprojects
 mediasoup-worker: setup
 ifeq ($(MEDIASOUP_WORKER_BIN),)
-	$(MESON) compile -j $(CORES) -C $(MEDIASOUP_OUT_DIR)/$(MEDIASOUP_BUILDTYPE) mediasoup-worker
+	$(MESON) compile -j $(CORES) $(MEDIASOUP_OUT_DIR)/$(MEDIASOUP_BUILDTYPE) mediasoup-worker
```
Why? `-C` specified build directory, you shouldn't remove it.

kcking replied:
Sorry I'm doing some local debugging -- the build is failing with git but working with a local path dependency. sorry for the noise ill close the PR for now
